src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java

```diff
@@ -762,46 +764,42 @@ private boolean checkAll(HttpServletRequest request, String cache, Nameable enti
             m = new TreeMap<>();
         } else if ((val = m.get(entity.getName())) != null) {
             // cache hit
-            stats.addRequest("authorization_cache_hits");
+            stats.increaseCounter("authorization_cache_hits");
             return val;
         }
 
-        stats.addRequest("authorization_cache_misses");
+        stats.increaseCounter("authorization_cache_misses");
+
+        Instant checkStart;
+        boolean overallDecision;
 
-        long time = 0;
-        boolean overallDecision = false;
-        
         lock.readLock().lock();
         try {
             // Make sure there is a HTTP session that corresponds to current plugin version.
             HttpSession session;
             if (((session = request.getSession(false)) != null) && isSessionInvalid(session)) {
                 session.invalidate();
-                stats.addRequest("authorization_sessions_invalidated");
+                stats.increaseCounter("authorization_sessions_invalidated");
             }
             request.getSession().setAttribute(SESSION_VERSION, getPluginVersion());
 
-            time = System.currentTimeMillis();
+            checkStart = Instant.now();
 
             overallDecision = performCheck(entity, pluginPredicate, skippingPredicate);
         } finally {
             lock.readLock().unlock();
         }
 
-        if (time > 0) {
-            time = System.currentTimeMillis() - time;
-
-            stats.addRequestTime("authorization", time);
-            stats.addRequestTime(
-                    String.format("authorization_%s", overallDecision ? "positive" : "negative"),
-                    time);
-            stats.addRequestTime(
-                    String.format("authorization_%s_of_%s", overallDecision ? "positive" : "negative", entity.getName()),
-                    time);
-            stats.addRequestTime(
-                    String.format("authorization_of_%s", entity.getName()),
-                    time);
-        }
+        Duration checkTime = Duration.between(checkStart, Instant.now());
```

Comment:
I wonder if this is faster than substracting two longs..

Reply:
It is definitely not faster, but I prefer readability and cleaner code than to save a few instructions.